src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -6,24 +6,21 @@
 
 public class BattleshipModel {
 
-
     //                  Member variables
+    protected Ship[] playerArray;
     protected Ship playerAircraftCarrier;
 
+    protected Ship[] computerArray;
     protected Ship computerAircraftCarrier;
 
-    // all ships for player or computer are collected in these lists
```

Comment:
(The Gson bug he is talking about is changing every ship in the array gets switched back to just Ship instead of Stealth_Ship/Civilian_Ships making so that you have to hard code putting in the individual ship into the array every time there is a call to the back end)